classes/helper.php

```diff
@@ -371,4 +374,91 @@ public static function regular_expression_search(string $search, string $table,
         }
         return $results;
     }
+
+    /**
+     * Get column info from column name.
+     *
+     * @param string $table The table name.
+     * @param string $columnname The column name.
+     *
+     * @return database_column_info|null The column info.
+     */
+    private static function get_column_info(string $table, string $columnname): ?database_column_info {
+        global $DB;
+
+        $columns = $DB->get_columns($table);
+        foreach ($columns as $col) {
+            if ($col->name == $columnname) {
+                return $col;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Replace all text in a table and column.
+     *
+     * @param string $table The table to search.
+     * @param string $columnname The column to search.
+     * @param string $search The text to search for.
+     * @param string $replace The text to replace with.
+     * @param int $id The id of the record to restrict the search.
+     */
+    public static function replace_text_in_a_record(string $table, string $columnname,
+                                                    string $search, string $replace, int $id) {
+
+        $column = self::get_column_info($table, $columnname);
+        self::replace_all_text($table, $column, $search, $replace, ' AND id = ?', [$id]);
```

Comment:
in the rest of moodle, the where clause should not start with AND to make using the api simpler